### PR TITLE
feat: added system notification

### PR DESCRIPTION
**Pull request by SunGoldTech**
## What type of PR is this? (check all applicable) 

- [x] 🍕 Feature

- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description

This PR adds the flash notification features...

## Related Tickets & Documents
feat  #281 
[https://github.com/open-sauced/hot/issues/281#issue-1326298617](https://github.com/open-sauced/hot/issues/281#issue-1326298617)

## Mobile & Desktop Screenshots/Recordings
after merge notifications will pop up on repo submit as shown below

> desktop error flash message on invalid repo submit

Screenshot (8)

> desktop success flash message on valid repo submit

Screenshot (9)

> mobile error flash message on invalid repo submit

Screenshot (12)

> mobile success flash message on valid repo submit

Screenshot (13)

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

**Review by 0-vortex — APPROVED** [review]
LGTM!

The toaster can be implemented into some other components and some error logging we are doing on beta but those can be done as separate issues! 🍕

**Comment by CMCSWEENE** [comment]
notifications work on desktop successful build

File: src/lib/reactHotToast.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import toast from "react-hot-toast";
+
+export declare interface ToastTriggerProps {
+  message: string;
+  type: "success" | "error" | "custom" | "";
+}
+
+// Notifier function trigger
+export const ToastTrigger = ({ message, type }: ToastTriggerProps) => {
```
Comment (NsdHSO) [inline src/lib/reactHotToast.ts:9]: suggestion: ToastTrigger -> toastTrigger

**Review by NsdHSO — COMMENTED** [review]
LGTM

**Comment by 0-vortex** [comment]
@bdougie any style suggestions for the toasters? should we have some defined error/warn/success styles?

File: src/lib/reactHotToast.ts (hunk 1 above)
Comment (bdougie) [reply src/lib/reactHotToast.ts:9]: agree, camelCase FTW

**Comment by bdougie** [comment]
> @bdougie any style suggestions for the toasters? should we have some defined error/warn/success styles?

My suggestion is to go with the default for now, mainly because I don't. The app.opensauced present errors and notifs in line with the form. My only style suggestion is to move the position to the bottom next to the submission form for now.

File: src/lib/reactHotToast.ts (hunk 1 above)
Comment (SunGoldTech) [reply src/lib/reactHotToast.ts:9]: Okay... I also concur :)

File: src/App.tsx [hunk 2]
```diff
@@ -8,24 +8,27 @@ import RepoSubmission from "./components/RepoSubmission";
...
-          {user && <RepoSubmission user={user} />}
-          <PrimaryNav/>
-          <Hero/>
-        </GradBackground>
-        <PostsWrap textToSearch={textToSearch} />
-        <Footer />
-      </div>
-    </BrowserRouter>
+    <>
+      <Toaster position={"top-center"} />
```
Comment (bdougie) [inline src/App.tsx:19]:
```suggestion
      <Toaster position={"top-right"} />
```

**Comment by bdougie** [comment]
Thanks for the contribution @SunGoldTech. I went ahead and committed what I suggested.

**Comment by bdougie** [comment]
Just an FYI for next time to add "closes #281" to the PR body when opening

**Comment by SunGoldTech** [comment]
> > @bdougie any style suggestions for the toasters? should we have some defined error/warn/success styles?
> 
> My suggestion is to go with the default for now, mainly because I don't. The app.opensauced present errors and notifs in line with the form. My only style suggestion is to move the position to the bottom next to the submission form for now.


Suggestions: I think having the toast at the top-center position looks good on all devices compared to the bottom as shown below


Screenshot (16)

Screenshot (13)

**Comment by bdougie** [comment]
We went with the top-right instead, but happy to take contribution to move it back if it presents bugs.

**Comment by SunGoldTech** [comment]
> We went with the top-right instead, but happy to take a contribution to move it back if it presents bugs.

Ok...That's cool

**Pull request merged**